### DISCUSSION TITLE
Add ability to compile zeek with mimalloc

### DISCUSSION THREAD

**Discussion by csbflyer** [zeek/zeek#1151] [How-Tos]
Please add the ability to compile with mimalloc similar to jemalloc

e.g.
./configure --enable-mimalloc

https://github.com/microsoft/mimalloc

**Comment by rsmmr** [comment]
Have you already managed to compile it against mimalloc by setting flags  through environment/configure? If so, could you share?

Generally, I'd prefer to have the right hooks through `configure` available to support such integrations generically, and then document how to do it for different libraries; in contrast to hardcoding more options. That's because experience shows that over time, new/better options keep coming up. In that spirit, I'd actually argue for removing the explicit jemalloc option, too.

**Comment by csbflyer** [comment]
mimalloc can be compiled with zeek using:

CFLAGS=’-L/path/to/mymalloc –lmymalloc’ CXXFLAGS=’-L/path/to/mymalloc
–lmymalloc’ ./configure

https://github.com/microsoft/mimalloc

I agree with your argument. Maybe Zeek can add a section in the
documentation on how to add a non-glibc( ptmalloc) malloc instead?

Based on research performed by Jim Mellander, the current top performing
mallocs are:
1. llalloc - http://locklessinc.com/downloads/
2. mimalloc - Microsoft
3. tcmalloc - Google
4. jemalloc - Facebook

**Comment by rsmmr** [comment]
I'll move this into the new HowTos section in Discussions.
